### DISCUSSION TITLE
Raspberry Pi 4: How to do VerticalSync on two displays on two different CPU cores?

### DISCUSSION THREAD

**Discussion by mktazo** [rsta2/circle#406] [Q&A]
Hi Rene,

fist of all I really appreciate your great work. It is so much fun to do bare metal programming on the Raspberry Pi using the fantastic Circle framework. Thanks a lot for creating and providing this.

I wonder if  and how it would be possible to create an application that uses Cores 2 and 3 to do graphics processing/screen update on the framebuffers for the two HDMI displays and synchronize core 2 to VerticalBlanc of the first display and core 3 to the second display. 

Is there a way to get an VerticalBlanc interrupt of these displays and route it to core 2 / core 3 -- may be using an Inter-Processor Interrupt (IPI)?

Greetings from Munich!
Best regards, Martin

**Comment by rsta2** [comment]
Hi Martin, thanks for appreciating Circle! Unfortunately it's only possible to wait for VSync on one display using `CBcmFrameBuffer::WaitForVerticalSync()`, because the property mailbox driver, which is used for this function, allows only one outstanding request at a time. So cannot wait for two events at once.

Rene

**Reply by mktazo** [reply]
Hi Rene, thanks for your explanation. CBcmFrameBuffer::WaitForVerticalSync() does a busy looping until VSync happened. Is there any possible way to let the VideoCore trigger an Interrupt on VSync, so the CPU can do more useful stuff?
Best Regards,
Martin

**Reply by rsta2** [reply]
There is a solution for the RPi 1-3, which is described in issue #50. There is also a test program attached. You have to add the following line to *kernel.cpp* for the RPi 4 to get it working there:

```
#define ARM_IRQ_SMI	GIC_SPI (112)
```
